### PR TITLE
🔥 remove posts.markdown field

### LINKED ISSUE #8479: Tidy up markdown usages
For `1.0` we have added a `mobiledoc` database field and a mobiledoc converter.
The new editor (see https://github.com/TryGhost/Ghost-Admin/commit/4b48328e6d7e8f7e2773c8cc006e9bcf2220547f) is using the new database field.

There is no need to keep the old markdown field, as well as the old legacy converter, see https://github.com/TryGhost/Ghost/blob/master/core/server/models/post.js#L236.

We would like to remove all markdown related code:

- [x] remove markdown field from the schema (https://github.com/TryGhost/Ghost/issues/8497)
- [x] remove markdown legacy converter (https://github.com/TryGhost/Ghost/issues/8497)
- [x] remove admin related markdown usages (https://github.com/TryGhost/Ghost-Admin/pull/720)

### PR DESCRIPTION
**Pull request by kevinansfield**
closes #8479
- removes `markdown` field from schema
- removes `legacyMarkdown` converter
- updates tests to work with `mobiledoc` field instead of `markdown` and adapt for mobiledoc HTML output where necessary

TODO:
- [x] find why the RSS page 2 test is failing
- [x] move duplicated `mobiledocDocument` functions into a test helper

### CODE REVIEW

File: core/test/integration/model/model_posts_spec.js [hunk 1]
```diff
@@ -457,20 +474,21 @@ describe('Post Model', function () {
...
-                    var post = results.toJSON();
-                    post.title.should.not.equal('123');
-                    return PostModel.edit({markdown: 123}, _.extend({}, context, {id: postId}));
-                }).then(function (edited) {
-                    should.exist(edited);
-                    edited.attributes.markdown.should.equal('123');
-                    done();
-                }).catch(done);
-            });
+            // TODO: is a similar test needed for mobiledoc?
```
Comment (kevinansfield) [inline core/test/integration/model/model_posts_spec.js]: @kirrg001 can this and the other commented test below be removed? The `mobiledoc` field can't be a plain number so I don't think these tests are relevant any more

File: core/test/functional/routes/api/posts_spec.js [hunk 2]
```diff
@@ -7,6 +7,23 @@ var should = require('should'),
     ghost = testUtils.startGhost,
     request;
 
+function mobiledocDocument(content) {
```
Comment (kevinansfield) [inline core/test/functional/routes/api/posts_spec.js]: I've duplicated this function a number of times - where would be the best place to define this as a test util?

File: core/test/utils/fixtures/export/export-000.json [hunk 3]
```diff
@@ -10,8 +10,8 @@
                 "uuid": "8492fbba-1102-4b53-8e3e-abe207952f0c",
                 "title": "Welcome to Ghost",
                 "slug": "welcome-to-ghost",
-                "markdown": "You're live! Nice.",
-                "html": "<p>You're live! Nice.</p>",
+                "mobiledoc": "{\"version\":\"0.3.1\",\"markups\":[],\"atoms\":[],\"cards\":[[\"card-markdown\",{\"cardName\":\"card-markdown\",\"markdown\":\"You're live! Nice.\"}]],\"sections\":[[10,0]]}",
```
Comment (kevinansfield) [inline core/test/utils/fixtures/export/export-000.json]: Should these export fixtures be using the 1.0 format or should they be kept in the LTS format? I noticed there were other LTS specific fixtures so I wasn't sure.

File: core/test/functional/routes/api/posts_spec.js (hunk 2 above)
Comment (kirrg001) [inline core/test/functional/routes/api/posts_spec.js]: There are two things which are annoying in the test env.
Some of the tests are not using the data-generator directly (`e.g. DataGenerator.createPost`) to generate post content and the test env doesn't use our models to insert data (it uses `knex` directly). But we want to change this in the future. Right now, this eats too much time.

For now, i would put this function to `test/utils/fixtures/data-generator.js`, which looks like a useful place.

Then you can access the function with `testUtils.DataGenerator.transformIntoMobiledoc` or similar.

And in the future, we should always use the `DataGenerator.createPost` helper, pass only html/plaintext, which is automatically transformed into mobiledoc by our models layer.

File: core/test/integration/model/model_posts_spec.js (hunk 1 above)
Comment (kirrg001) [inline core/test/integration/model/model_posts_spec.js]: 👍 remove

File: core/test/utils/fixtures/export/export-000.json (hunk 3 above)
Comment (kirrg001) [inline core/test/utils/fixtures/export/export-000.json]: LTS

Comment (kevinansfield) [inline core/test/utils/fixtures/export/export-000.json]: 👍 reverted

File: core/test/integration/model/model_posts_spec.js [hunk 4]
```diff
@@ -64,8 +65,8 @@ describe('Post Model', function () {
 
             // @TODO change / update this for mobiledoc in
```
Comment (kirrg001) [inline core/test/integration/model/model_posts_spec.js]: I think we can remove the comment.

Comment (kevinansfield) [inline core/test/integration/model/model_posts_spec.js]: done

**Review by kirrg001 — APPROVED** [review]
LGTM 👍 
Thank you!

**Pull request merged**